nncf/quantization/algorithms/weight_compression/weight_lowering/dispatcher.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2024 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from enum import Enum
+from functools import wraps
+from typing import Any, Callable, Dict, Optional
+
+from nncf.utils import is_openvino_available
+
+
+class WeightLoweringBackend(Enum):
+    TENSOR = "TENSOR"
+    OV = "OV"
+
+
+def ov_available_backend_selector(*args, **kwargs) -> WeightLoweringBackend:
+    forced_backend = kwargs.get("forced_backend", None)
+    if forced_backend is not None:
+        return forced_backend
+    return WeightLoweringBackend.OV if is_openvino_available() else WeightLoweringBackend.TENSOR
+
+
+# Dispatcher decorator factory that accepts a backend selector function
+def weight_lowering_dispatcher(backend_selector_fn: Callable):
```

Comment:
Honestly, this is very difficult implementation for selecting calling 2 functions. Could you think how you can simplify this?